code/game/objects/items/blink_drive.dm

```diff
@@ -0,0 +1,169 @@
+#define BLINK_DRIVE_RANGE 7
+#define BLINK_DRIVE_MAX_CHARGES 3
+#define BLINK_DRIVE_CHARGE_TIME 2 SECONDS
+
+/obj/item/blink_drive
+	name = "blink drive"
+	desc = "A portable Bluespace Displacement Drive, otherwise known as a blink drive. Can teleport the user across short distances with a degree of unreliability, with potentially fatal results. <b>Alt right click or middleclick to teleport to a destination when the blink drive is equipped.</b>"
+	icon = 'icons/obj/items/jetpack.dmi'
+	icon_state = "bluespace_pack"
+	w_class = WEIGHT_CLASS_BULKY
+	flags_equip_slot = ITEM_SLOT_BACK
+	obj_flags = CAN_BE_HIT
+	///Number of teleport charges you currently have
+	var/charges = 3
+	///True if you can use shift click/middle click to use it
+	var/selected = FALSE
+	///The timer for recharging the drive
+	var/charge_timer
+	///The mob wearing the blink drive. Needed for item updates.
+	var/mob/equipped_user
+	COOLDOWN_DECLARE(blink_stability_cooldown)
+
+/obj/item/blink_drive/update_icon()
+	. = ..()
+	if(equipped_user)
+		equipped_user.update_inv_back()
```

Comment:
equipped_user?.update

basically same but in one line

Reply:
done